solver/llbsolver/solver.go

```diff
@@ -222,7 +226,10 @@ func (s *Solver) recordBuildHistory(ctx context.Context, id string, req frontend
 			}
 		}
 
-		makeProvenance := func(res solver.ResultProxy, cap *provenance.Capture) (*controlapi.Descriptor, func(), error) {
+		makeProvenance := func(name string, res solver.ResultProxy, cap *provenance.Capture) (*controlapi.Descriptor, func(), error) {
+			span, ctx := tracing.StartSpan(ctx, fmt.Sprintf("create %s history provenance", name))
```

Comment:
Can `name` be an attribute on the span instead of part of the name?

Reply:
In the Jaeger UI, spans need individual names to pick the correct line. Other details, like attributes, become visible only when a span is clicked on.